app/controllers/DetermineTaxIdController.scala

```diff
@@ -55,7 +55,11 @@ class DetermineTaxIdController @Inject() (
   def determineTaxId(): Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
     val (maybeTaxId, taxRegime): (Future[Option[TaxId]], TaxRegime) = request.journey match {
       case j: Journey.Stages.Started =>
-        enrolmentService.determineTaxIdAndUpdateJourney(j, request.enrolments) -> j.taxRegime
+        if (j.taxRegime == TaxRegime.Sia) {
+          (Future(request.nino.asInstanceOf[Option[TaxId]]), j.taxRegime)
```

Comment:
isn't `Nino` already a `TaxId`? Do you need this `.asInstanceOf`?

Reply:
originally it was complaining for not doinf the instance of, but you're right, doesnt need to be there